backend/test/services/LibraryService.test.ts

```diff
@@ -180,6 +180,74 @@ describe('LibraryService', () => {
     });
   });
 
+  describe('create', () => {
+    it('returns 201 status with populated location when provided correct headers and a FHIR Library', async () => {
+      await supertest(server.app)
+        .post('/4_0_1/Library')
+        .send({ resourceType: 'Library', status: 'draft' })
+        .set('content-type', 'application/json+fhir')
+        .expect(201)
+        .then(response => {
+          expect(response.headers.location).toBeDefined();
+        });
+    });
+  });
+
+  describe('update', () => {
+    it('returns 200 when provided correct headers and a FHIR Library whose id is in the database', async () => {
+      await createTestResource(
+        { resourceType: 'Library', type: { coding: [{ code: 'logic-library' }] }, id: 'exampleId', status: 'draft' },
+        'Library'
+      );
+      await supertest(server.app)
+        .put('/4_0_1/Library/exampleId')
+        .send({ resourceType: 'Library', id: 'exampleId', status: 'active' })
+        .set('content-type', 'application/json+fhir')
+        .expect(200);
+      await supertest(server.app)
+        .get('/4_0_1/Library/exampleId')
+        .expect(200)
+        .then(response => {
+          expect(response.body.id).toEqual('exampleId');
+        });
+    });
+
+    it('returns 201 when provided correct headers and a FHIR Library whose id is not in the database', async () => {
+      await supertest(server.app)
+        .put('/4_0_1/Library/newId')
+        .send({ resourceType: 'Library', id: 'newId', status: 'draft' })
+        .set('content-type', 'application/json+fhir')
+        .expect(201)
+        .then(response => {
+          expect(response.headers.location).toBeDefined();
+        });
+      await supertest(server.app)
+        .get('/4_0_1/Library/newId')
+        .expect(200)
+        .then(response => {
+          expect(response.body.resourceType).toEqual('Library');
+          expect(response.body.id).toEqual('newId');
+          expect(response.body.status).toEqual('draft');
+        });
+    });
```

Comment:
Same here, this seems like getting a bit close to integration testing. I think if we can test that the headers and status code are what we expect, we can trust mongo and asymmetrik to work as advertised in terms of actually storing the resource and making it accessible.